### PR TITLE
UI Enhancements, Role-Based Access Control, and New Payment Features

### PR DESCRIPTION
**Pull request by bilalmkhalil**
1. Added a checkbox for admin and support in the "Add Sub Admin" form.
2. Changed the UI of the profile dropdown to match the Figma design.
3. Fixed sidebar navigation icon color inconsistency and active state.
4. Added guideline navigation with nested items.
5. Added active support chat highlighting.
6. Changed checkbox design according to Figma.
7. Made search input functional in chat support to search through the list.
8. Added a new payment page:
      - List of withdrawal requests.
      - Button in the table to accept withdrawal requests.
      - Displaying pending, available, and goollooper amounts at the top of the table.
      - Added a withdrawal button next to the goollooper amount.
9. Created and modified a new confirmation modal using conditional rendering.
10. Fixed height issues on some pages, ensuring consistent bottom margins regardless of screen height.
11. Added input fields in interest page and volunteer page same as in Figma.
12. Redesigned the toolbar of the editor to match the Figma design.
13. Added a filter dropdown to filter users by user type and service provider.
14. Fixed the UI for sign in, forgot password, and reset password. Also fixed all layouts, margins, padding, and font sizes for the dashboard, users, sub-admins, support, guidelines, and push notifications.
15. Fixed an issue where users could access pages via URL without being logged in as admin or any role.
16. Added guard role functionality where sub-admins cannot access the payment page, and the support role can only access the support page. These pages will not appear in the sidebar, nor can they be accessed via URL.

### CODE REVIEW

File: app/dashboard/page.tsx
Comment (xNOVA2) [inline app/dashboard/page.tsx:1]: its better to make setLoading false in finally block

File: api/index.tsx [hunk 1]
```diff
@@ -49,15 +49,18 @@ export const getUsers = function (
   email?: string,
   role?: number | null,
 ) {
-  return Api.get(`/admin/user?page=${page}&limit=${limit}&email=${email || ""}&role=${role || ""}`);
+  const emailQuery = email ? `&email=${email}` : '';
+  const roleQuery = role !== null && role !== undefined ? `&role=${role}` : '';
```
Comment (bilalmkhalil) [inline api/index.tsx:53]: made email query optional

File: api/index.tsx [hunk 2]
```diff
@@ -49,15 +49,18 @@ export const getUsers = function (
...
+  return Api.get(`/admin/user?page=${page}&limit=${limit}${emailQuery}${roleQuery}`);
 };
 
 export const getSubadmin = function (
   page: number = 1,
   limit: number = 10,
   email?: string
 ) {
-  return Api.get(`/admin/sub-admin?page=${page}&limit=${limit}&email=${email}`);
+  const emailQuery = email ? `&email=${email}` : '';
```
Comment (bilalmkhalil) [inline api/index.tsx:62]: made email query option in fetching subadmins

**Comment by xNOVA2** [comment]
**Nice work bilal**

**Pull request merged**